include/picongpu/plugins/Emittance.hpp

```diff
@@ -0,0 +1,1050 @@
+/* Copyright 2013-2018 Axel Huebl, Felix Schmitt, Heiko Burau,
+ *                     Rene Widera, Richard Pausch, Benjamin Worpitz,
+ *                     Sophie Rudat
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/simulation_defines.hpp"
+#include "picongpu/algorithms/KinEnergy.hpp"
+#include "picongpu/plugins/common/txtFileHandling.hpp"
+#include "picongpu/plugins/multi/multi.hpp"
+#include "picongpu/particles/traits/SpeciesEligibleForSolver.hpp"
+#include "picongpu/particles/traits/GenerateSolversIfSpeciesEligible.hpp"
+#include "picongpu/plugins/misc/misc.hpp"
+#include "picongpu/simulationControl/MovingWindow.hpp"
+
+#include <pmacc/mappings/kernel/AreaMapping.hpp>
+#include <pmacc/cuSTL/algorithm/mpi/Gather.hpp>
+#include <pmacc/cuSTL/algorithm/mpi/Reduce.hpp>
+#include <pmacc/mpi/MPIReduce.hpp>
+#include <pmacc/nvidia/functors/Add.hpp>
+#include <pmacc/cuSTL/algorithm/functor/Add.hpp>
+#include <pmacc/memory/shared/Allocate.hpp>
+#include <pmacc/dataManagement/DataConnector.hpp>
+#include <pmacc/nvidia/atomic.hpp>
+#include <pmacc/mappings/threads/ForEachIdx.hpp>
+#include <pmacc/mappings/threads/IdxConfig.hpp>
+#include <pmacc/memory/CtxArray.hpp>
+#include <pmacc/traits/GetNumWorkers.hpp>
+#include <pmacc/traits/HasIdentifiers.hpp>
+#include <pmacc/traits/HasFlag.hpp>
+#include <pmacc/algorithms/ForEach.hpp>
+#include <pmacc/math/Vector.hpp>
+
+#include <boost/mpl/and.hpp>
+
+#include <vector>
+#include <algorithm>
+#include <utility>
+#include <string>
+#include <iostream>
+#include <fstream>
+#include <memory>
+#include <stdexcept>
+
+
+namespace picongpu
+{
+
+    /** calculates the emittance in x direction
```

Comment:
`... in y direction`

Reply:
in x direction is correct but along y - I will add a comment